_studio/mfx_lib/mctf_package/mctf/src/mctf_common.cpp

```diff
@@ -3160,17 +3163,11 @@ mfxU32 CMC::IM_SURF_PUT(
 }
 
 mfxU32 CMC::IM_SURF_PUT(
-    CmSurface2D *p_DstSurface,
-    CmSurface2D *p_SrcSurface)
+    CmSurface2D **p_DstSurface,
+    CmSurface2D **p_SrcSurface)
 {
-    CM_STATUS
-        status = CM_STATUS_FLUSHED;
-    res = queue->EnqueueCopyGPUToGPU(p_DstSurface, p_SrcSurface, NULL, copyEv);
-    MCTF_CHECK_CM_ERR(res, res);
-    copyEv->GetStatus(status);
-    while (status != CM_STATUS_FINISHED)
-        copyEv->GetStatus(status);
-    return res;
+    *p_DstSurface = *p_SrcSurface;
```

Comment:
hm. Do I understand it correctly that we still rewrite encoder input surface with the mctf output? If not I am not sure what you are doing here and why it is needed? (and I don't see memory allocation for the mctf output surface...)

Reply:
This function is going to be removed.